### PR TITLE
Design review: gatsby-theme-docs tasks that can be combined

### PR DESCRIPTION
**Pull request by davifantasia**
Closes https://github.com/commercetools/commercetools-docs-kit/issues/565

### CODE REVIEW

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 1]
```diff
@@ -140,85 +143,110 @@ const SearchContainer = styled.div`
...
-              onClick={props.closeTopMenu}
+            <SwitcherButton
+              role="button"
+              aria-label={
+                props.isTopMenuOpen ? 'Close Top Menu' : 'Open Top Menu'
+              }
+              isActive={props.isTopMenuOpen}
+              onClick={props.toggleTopMenu}
+              onMouseOver={handleOnMouseOverTopMenu}
+              onMouseLeave={handleOnMouseLeaveTopMenu}
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:182]: Why do we need mouse events? We should be able to do this with CSS only.

Comment (davifantasia) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:182]: I don't have a solution for css only. Do you have any suggestion?

Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:182]: I can take a look later but I would prefer to not use event listeners if not necessary.

Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-header.js]: I pushed a commit with the changes: https://github.com/commercetools/commercetools-docs-kit/pull/581/commits/af399b88b5a17790f82360aec1bcb206258817f3

I also took the chance to improve the a11y of the button element (see gif, where I use tab navigation to focus on the element and then toggle the menu using the spacebar and the enter keys).

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 2]
```diff
@@ -116,17 +120,21 @@ const DocumentationSwitcherContainer = styled.div`
...
+  :hover,
+  :focus {
+    border-bottom: 2px solid ${designSystem.colors.light.linkNavigation};
+    color: ${designSystem.colors.light.linkNavigation};
+    svg {
+      * {
+        fill: ${designSystem.colors.light.linkNavigation};
+      }
+    }
+  }
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:132]: This is the CSS part to handle hover and focus effects.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 3]
```diff
@@ -116,17 +120,21 @@ const DocumentationSwitcherContainer = styled.div`
...
+
   @media screen and (${designSystem.dimensions.viewports.desktop}) {
     margin: 0 0 0 ${designSystem.dimensions.spacings.xl};
   }
 `;
-const SwitcherButton = styled.a`
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:123]: This is not necessary anymore, I "merged" it into the `DocumentationSwitcherContainer`, which is now `DocumentationSwitcherButton`

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 4]
```diff
@@ -140,31 +148,48 @@ const SearchContainer = styled.div`
...
+      event.key === 'Enter' ||
+      event.key === ' ' ||
+      event.key === 'Spacebar' ||
+      event.which === 13 ||
+      event.which === 32;
+    if (enterOrSpace) {
+      event.preventDefault();
+      props.toggleTopMenu(event);
+    }
+  };
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:163]: To handle the spacebar and enter keys for a11y.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-header.js [hunk 5]
```diff
@@ -140,31 +148,48 @@ const SearchContainer = styled.div`
...
+                  display: none;
+                }
+              `}
+            />
+            <LogoButton />
+          </LogoContainer>
+          <DocumentationSwitcherButton
             role="button"
+            id="top-menu-switcher"
+            tabIndex="1"
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-header.js:185]: To make the element accessible

**Comment by emmenko** [comment]
@davifantasia can you check the API e2e tests?

**Pull request merged**